### ISSUE TITLE
Wrong warning? integer / 1 "Operation will always return the left side of the expression."

### ISSUE THREAD

**Issue by PragTob** [rrrene/credo#104]
In my project credo complains:

```
tobi@airship ~/github/benchee $ mix credo
Checking 6 source files ...

  Warnings - please take a look                                                      
┃ 
┃ [W] ↗ Operation will always return the left side of the expression.
┃       lib/benchee/statistics.ex:121 (Benchee.Statistics.to_float)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.1 seconds (0.00s to load, 0.1s running checks)
34 mods/funs, found 1 warning.

Showing priority issues: ↑ ↗ →  (use `--strict` to show all issues, `--help` for options).
```

The offending function is:

```elixir
  defp to_float(maybe_integer) do
    maybe_integer / 1
  end
```

Which is the only way I found to convert an integer to a float (couldn't find `to_float` or anything in stdlib.) E.g. I need to be sure to have a float when I hand it to `:io_lib.format/2`. Also it doesn't return the left side, the left side as a float. So I guess I'm not doing anything wrong here, or am I?

I use 4.0-beta4.

If you want to have a full look, the repository is: https://github.com/PragTob/benchee

edit: Almost forgot to say, thanks for credo! Wonderful project! :heart:

**Comment by rrrene** [comment]
I would use the following (not sure if it is the idiomatic way):

    defp to_float(maybe_integer) do
      {float, _rest} = maybe_integer |> to_string  |> Float.parse
      float
    end

This way `value` can be a binary, an integer or a float and you always get a float. Only downside of this implemenation is that you have to be sure one of these conditions applies. If you want to error-check the results, you can use [Float.parse/1](http://elixir-lang.org/docs/stable/elixir/Float.html#parse/1) directly.

That said, I did not know of the way you solved it. It is interesting, especially when coming from Ruby where division by 1 would always result in an integer.

**Comment by PragTob** [comment]
yeah found that interesting in Elixir as well, apparently `/` always results in a float otherwise you gotta do `Kernel.div/2`. 

Your approach is interesting, but for me it'll never be a binary (it comes straight out of a statistics calculation module) + parsing something that I already had seems weird.

Thanks for the work and the reply!

**Comment by requnix** [comment]
I have this same issue. In my code I have a value that is always an integer and I need it to be converted to a float for which I'm using `value / 1` which seems to be the Elixir way.

This shows up as a warning, which is kind of annoying. I definitely don't want to add binary processing and defensive programming to get rid of a warning.

**Comment by rrrene** [comment]
`v0.4.5` changes the behaviour. After re-reading the source of `Elixir.Float` if had to agree that this check does more harm than good :+1:

**Comment by joerixaop** [comment]
`:erlang.float/1` works and is perhaps clearer in showing the intent.

**Comment by rrrene** [comment]
@joerixaop thx for sharing!

If anyone in this discussion is interested: http://erlang.org/doc/man/erlang.html#float-1 

In mind this would probably be the "BEAM way" to do it. :grin:

**Comment by PragTob** [comment]
seems to be the better way to do it, didn't know - thanks.
